server/src/main/java/org/opensearch/extensions/ExtensionActionListener.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.util.ArrayList;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction.Response;
+
+/**
+ * ActionListener = for ExtensionsOrchestratore
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListener<ExtensionBooleanResponse> implements ActionListener<Response> {
+
+    private static final Logger logger = LogManager.getLogger(ExtensionActionListener.class);
+    private int responseCount;
+    private ArrayList<Exception> exceptionList;
+
+    public ExtensionActionListener() {
+        responseCount = 0;
```

Comment:
Why do we need a counter here? We are not using it anywhere

Reply:
That's what AD does.   I'm not sure where it's used elsewhere, but I'm emulating as closely as possible to their functionality.

Reply:
Can you point any reference to the counter in AD?

Reply:
> We are not using it anywhere

We are using it in the getter below :-)

But seriously... I do see a counter in the MultiResponse.  Note it's using an `AtomicLong` to be thread safe, this implementation is not thread safe, and not sure it's needed in the general case.

Reply:
If I was not clear, I meant we are not using the `getter` as well anywhere.